pkg/rhcos/cache/cache.go

```diff
@@ -262,7 +262,7 @@ func (u *urlWithIntegrity) download(dataType, applicationName string) (string, e
 // DownloadImageFile is a helper function that obtains an image file from a given URL,
 // puts it in the cache and returns the local file path.  If the file is compressed
 // by a known compressor, the file is uncompressed prior to being returned.
-func DownloadImageFile(baseURL string, applicationName string) (string, error) {
+func DownloadImageFile(baseURL string, applicationName string, sha256Checksum string) (string, error) {
```

Comment:
To avoid impacting the code in several places where this additional check is not required, it could be worth to write an additional (ie just an example) `DownloadImageFileWithSha` function with the additional parameter, and use it in the only place where it's required (leaving the rest untouched)

Reply:
Yeah, I don't really like having similar functions as there's not a good way to reduce code but I also didn't like changing the function signature. I'll revisit this.